test/modules/candy-machine/initCandyMachine.test.ts

```diff
@@ -0,0 +1,371 @@
+import test from 'tape';
+import {
+  amman,
+  assertConfirmedWithoutError,
+  assertThrows,
+  hash32Bit,
+  killStuckProcess,
+  metaplex,
+  SKIP_PREFLIGHT,
+  spokSameBignum,
+  spokSamePubkey,
+} from '../../helpers';
+import {
+  CandyMachine,
+  cusper,
+  EndSettingType,
+  GatekeeperConfig,
+  WhitelistMintMode,
+  WhitelistMintSettings,
+} from '@metaplex-foundation/mpl-candy-machine';
+import { Keypair, PublicKey } from '@solana/web3.js';
+import spok from 'spok';
+import { Signer } from '../../../src/shared';
+import {
+  CandyMachineConfigWithoutStorage,
+  creatorsConfigDefault,
+} from '../../../src/modules/candy-machine/models/config';
+import { assertCreators } from '../../helpers/candy-machine';
+
+killStuckProcess();
+
+async function init() {
+  const mx = await metaplex();
+  const payer = mx.identity();
+
+  const solTreasuryAccount = Keypair.generate();
+  await amman.airdrop(mx.connection, solTreasuryAccount.publicKey, 100);
+
+  const minimalConfig: CandyMachineConfigWithoutStorage = {
+    price: 1.0,
+    number: 10,
+    sellerFeeBasisPoints: 0,
+    solTreasuryAccount: solTreasuryAccount.publicKey.toBase58(),
+    goLiveDate: '25 Dec 2021 00:00:00 GMT',
+    retainAuthority: true,
+    isMutable: false,
+  };
+
+  const opts = {
+    candyMachine: Keypair.generate(),
+    confirmOptions: SKIP_PREFLIGHT,
+  };
+  await amman.addr.addLabels({ ...minimalConfig, ...opts, payer });
+
+  const cm = mx.candyMachine();
+  return { cm, payer, solTreasuryAccount, minimalConfig, opts };
+}
+
+function assertProperlyInitialized(
+  t: test.Test,
+  {
+    candyMachineSigner,
+    payer: _,
+    wallet,
+    authority,
+    candyMachine,
+    price,
+    sellerFeeBasisPoints,
+    number,
+    isMutable,
+    retainAuthority,
+    goLiveDate,
+    tokenMint,
+  }: {
+    // Accounts
+    candyMachineSigner: Signer;
+    payer: Signer;
+    wallet: PublicKey;
+    authority: PublicKey;
+    // Data
+    candyMachine: CandyMachine;
+    // Config
+    price: number;
+    sellerFeeBasisPoints: number;
+    number: number;
+    isMutable: boolean;
+    retainAuthority: boolean;
+    goLiveDate: string;
+    // Extra
+    tokenMint: PublicKey | null;
+  }
+) {
+  spok(t, candyMachine.pretty(), {
+    $topic: 'candy machine',
+    authority: authority.toBase58(),
+    wallet: wallet.toBase58(),
+    tokenMint,
+    itemsRedeemed: spokSameBignum(0),
+    data: {
+      uuid: candyMachineSigner.publicKey.toBase58().slice(0, 6),
+      price: spokSameBignum(price),
+      sellerFeeBasisPoints: spokSameBignum(sellerFeeBasisPoints),
+      maxSupply: spokSameBignum(number),
+      itemsAvailable: spokSameBignum(number),
+      isMutable,
+      retainAuthority,
+      goLiveDate: spokSameBignum(new Date(goLiveDate).valueOf()),
+    },
+  });
+}
+
+// -----------------
+// Minimal Config
+// -----------------
+test('candyMachine: init with minimal config', async (t) => {
```

Comment:
Should we also prefix other tests?
Not sure what's the best convention with tape.

Reply:
I like to prefix with the main topic or file name without extension.
It's my convention to quickly find failing tests, but we can do whatever we want.
Tape gives us all the freedom.